integration-cli/docker_cli_inspect_test.go

```diff
@@ -38,6 +38,95 @@ func (s *DockerSuite) TestInspectInt64(c *check.C) {
 	}
 }
 
+func (s *DockerSuite) TestInspectDefault(c *check.C) {
+
+	//Both the container and image are named busybox. docker inspect will fetch the container JSON.
+
+	runCmd := exec.Command(dockerBinary, "run", "--name=busybox", "-d", "busybox", "true")
+	out, _, _, err := runCommandWithStdoutStderr(runCmd)
+	if err != nil {
+		c.Fatalf("failed to run container: %v, output: %q", err, out)
+	}
+
+	inspectCmd := exec.Command(dockerBinary, "inspect", "busybox")
+
+	_, exitCode, err := runCommandWithOutput(inspectCmd)
+	if exitCode != 0 || err != nil {
+		c.Fatalf("failed to inspect container: %s, %v", out, err)
+	}
+
+}
+
+func (s *DockerSuite) TestInspectTypeFlagContainer(c *check.C) {
+
+	/*
+	   Both the container and image are named busybox. docker inspect will fetch container
+	   JSON State.Running field. If the field is true, it's a container.
+	*/
+
+	runCmd := exec.Command(dockerBinary, "run", "--name=busybox", "-d", "busybox", "true")
+	out, _, _, err := runCommandWithStdoutStderr(runCmd)
+	if err != nil {
+		c.Fatalf("failed to run container: %v, output: %q", err, out)
+	}
+
+	formatStr := fmt.Sprintf("--format='{{.State.Running}}'")
+	inspectCmd := exec.Command(dockerBinary, "inspect", "--type=container", formatStr, "busybox")
+
+	out, exitCode, err := runCommandWithOutput(inspectCmd)
+	if exitCode != 0 || err != nil {
+		c.Fatalf("failed to inspect container: %s, %v", out, err)
+	}
+
+	if out != "true\n" {
+		c.Fatal("not a container JSON")
+	}
```

Comment:
as part of this test I think it would be good to add:
- another variant that uses "--type=image" to make sure it doesn't grab the container with the same name
- another variant that doesn't use "--type" at all to make sure it picks up the container